src/TestSuite/TestCase.php

```diff
@@ -249,9 +238,45 @@ protected function tearDown(): void
         $this->_configure = [];
         $this->_tableLocator = null;
 
-        if ($this->fixtureManager) {
-            $this->fixtureManager->teardownTest($this);
-            $this->fixtureManager = null;
+        $this->teardownFixtures();
+    }
+
+    /**
+     * Initialized and loads any use fixtures.
+     *
+     * @return void
+     */
+    protected function setupFixtures(): void
+    {
+        $fixtureNames = $this->getFixtures();
+        if (empty($fixtureNames)) {
+            return;
+        }
+
+        static::$fixtureStrategy = new $this->fixtureStrategyClass();
```

Comment:
Should this continue to come from a protected method? Having a method to create the strategy will help users override it more easily. We wouldn't have to check the return type that way either as we could use a return type hint to enforce the interface requirement.

Reply:
Avoiding the class property would be nice.